### ISSUE TITLE
Webpack refactor for production

### ISSUE THREAD

**Issue by natew** [motion/orbit#999]
I've been looking over it for the last day or so trying to just gather in all the requirements and current state of the app. It's not going to be an easy change as it requires a few things.

Right now we have an architecture where we have a webpack DLL, with one being all the apps, and the other being Orbit itself. Then for one-off dev apps we have an alternate system that andrey built via `AppMiddleware.tsx`.

What needs to happen is the two need to basically come together, and both change quite a bit.

Two big problems:

1. We are writing a file `appDefinitions.js` inside the `orbit-app` directory and then requiring it via `orbit-app/src/apps/orbitApps.ts`.

This needs to change, because in production we need to load them independently. But, we still want a DLL that both the `orbit-app` and the workspace apps both share. So, we'll need to create a third DLL, the "base" DLL, that has all the shared libraries inside it.

So whereas it is this now:

- apps DLL => main DLL

It will become:

- base DLL (@o/kit, @o/ui, etc) =>
- apps DLL, main DLL

And we'll need the main DLL to load apps a bit differently, likely more like how AppMiddleware/andrey solution does by exporting a global variable. That will likely then require some work to get it to HMR properly.

2. A big architectural mess in terms of how everything works together

We basically have the CLI which uses Bonjour to start the main orbit.app process (orbit-desktop and others). Then orbit-desktop goes back and runs the CLI again to run things. This is pretty messy. The desire here comes from my want to have the CLI be an external, non-bundled thing, not included in the Orbit.app, so we can upgrade it much more easily without needing to upgrade their entire .app. But perhaps this is actually not desirable, for two reasons:

- It would be more brittle, what if they upgrade the cli and an old orbit.app doesn't work with it well?
- It creates a lot of duplication and confusion generally, as they share a lot of code.

So basically I think resolving this means we just don't have the CLI be much of it's own separate thing, and instead we put most things inside orbit-desktop, and just run them through the CLI as needed.

This refactor would look like:

- [x] Move BuildServer into desktop.
- [x] Move WorkspaceManager into desktop.
- [x] Move many of the CLI helper functions into desktop too.
- [x] Refactor the above + AppsManager all into one area.
- [x] Make CLI call into desktop to do most things rather than doing them inside the cli itself.
